docs/porting.md

```diff
@@ -119,6 +119,14 @@ Opening an endpoint is done for all non-control endpoints once the host picks a
 
 Also make sure to enable endpoint specific interrupts.
 
+##### dcd_edpt_close
+
+Close an endpoint. his function is used for implementing alternate settings.
+
+After calling this, the device should not respond to any packets directed towards this endpoint. When called, this function must abort any transfers in progress through this endpoint, before returning.
+
+Implementation is optional. Must be called from the USB task. Interrupts could be disabled or enabled during the call.
```

Comment:
We actually don't have to tell where it is called or the context it is called. The only layer could access dcd API is usbd. (OK I know there is a direct class-to-dcd somewhere, it is a hack, will fix it later on :D )

Reply:
I think the edpt_open calls break the separation, as they are called directly from the class drivers.

But, yeah, we should just assume only the USB task can call the dcd_* functions. On the other hand, maybe a comment like this will prevent a bug in the future?

Reply:
> I think the edpt_open calls break the separation, as they are called directly from the class drivers.

That is the one, I am too lazy for that changes.

> But, yeah, we should just assume only the USB task can call the dcd_* functions. On the other hand, maybe a comment like this will prevent a bug in the future?

`Interrupts could be disabled or enabled during the call` I am not so sure about this, I didn't see the need for this just now. When there is actually need, we could assert this once.